extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -299,13 +349,145 @@ export class AmpStoryQuiz extends AMP.BaseElement {
    * @private
    */
   handleOptionSelection_(optionEl) {
-    this.triggerAnalytics_(optionEl);
+    this.responseDataPromise_.then(() => {
+      this.triggerAnalytics_(optionEl);
 
-    this.mutateElement(() => {
-      optionEl.classList.add('i-amphtml-story-quiz-option-selected');
-      this.quizEl_.classList.add('i-amphtml-story-quiz-post-selection');
+      this.mutateElement(() => {
+        optionEl.classList.add('i-amphtml-story-quiz-option-selected');
+        this.quizEl_.classList.add('i-amphtml-story-quiz-post-selection');
+
+        this.hasUserSelection_ = true;
+      });
+
+      this.updateReactionData_(optionEl.optionIndex_);
+    });
+  }
+
+  /**
+   * Get the Reaction data from the datastore
+   *
+   * @return {Promise<JsonObject>}
+   * @private
+   */
+  retrieveReactionData_() {
+    return this.executeReactionRequest_()
+      .then(response => {
+        if (response.hasNoEndpoint) {
+          return;
+        }
+
+        this.handleSuccessfulDataRetrieval_(response);
+      })
+      .catch(error => {
+        dev().error(TAG, error);
+      });
+  }
+
+  /**
+   * Update the Reaction data in the datastore
+   *
+   * @param {number} reactionValue
+   * @private
+   */
+  updateReactionData_(reactionValue) {
+    this.executeReactionRequest_(reactionValue).catch(error => {
+      dev().error(TAG, error);
+    });
+  }
+
+  /**
+   * Executes a Reactions API call.
+   *
+   * @param {number} reactionValue
+   * @return {Promise<JsonObject>}
+   * @private
+   */
+  executeReactionRequest_(reactionValue) {
+    // TODO(jackbsteinberg): Add a default reactions endpoint.
+    if (!this.element.hasAttribute('endpoint')) {
+      return Promise.resolve({hasNoEndpoint: true});
+    }
+
+    let URL = this.element.getAttribute('endpoint');
+
+    const requestVars = dict({
+      'reactionType': STORY_REACTION_TYPE_QUIZ,
+      'reactionId': `page=${this.storeService_.get(
+        StateProperty.CURRENT_PAGE_ID
+      )}&url=${this.win.location.href}`,
+    });
+
+    const requestOptions = dict({
+      'method': 'GET',
+    });
+
+    if (reactionValue !== undefined) {
+      requestVars['reactionValue'] = reactionValue;
+      requestOptions['method'] = 'POST';
+      requestOptions['body'] = requestVars;
+    } else {
+      URL += `?reactionType=${
+        requestVars['reactionType']
+      }&reactionId=${encodeURIComponent(requestVars['reactionId'])}`;
+    }
```

Comment:
Can you use the helpers we have at https://github.com/ampproject/amphtml/blob/master/src/url.js? They're helpful and might solve a few bugs, for example, the provided endpoint could already have some GET parameters, and you'd need to start with a `&` not a `?`.